src/controller/python/chip/clusters/attribute.cpp

```diff
@@ -352,35 +355,30 @@ PyChipError pychip_WriteClient_WriteAttributes(void * appContext, DeviceProxy *
         app::InteractionModelEngine::GetInstance()->GetExchangeManager(), callback->GetChunkedCallback(),
         timedWriteTimeoutMs != 0 ? Optional<uint16_t>(timedWriteTimeoutMs) : Optional<uint16_t>::Missing());
 
-    va_list args;
-    va_start(args, n);
-
     VerifyOrExit(device != nullptr && device->GetSecureSession().HasValue(), err = CHIP_ERROR_MISSING_SECURE_SESSION);
 
+    for (size_t i = 0; i < attributeDataLength; i++)
     {
-        for (size_t i = 0; i < n; i++)
+        void * path   = writeAttributesData[i].mAttributePath;
+        void * tlv    = writeAttributesData[i].mTlvData;
+        size_t length = writeAttributesData[i].mTlvLength;
+
+        python::AttributePath pathObj;
+        memcpy(&pathObj, path, sizeof(python::AttributePath));
```

Comment:
I find it somewhat confusing that we use python::AttributePath which is different than python::Controller::AttributePath (the latter having version) from data build from a sconstruct that is AttributePathIBStruct.

How can we guarantee this is ok ? IBStruct has data version however python::AttributePath has it. We probably do not care, however binary compatibility enforcement here is a bit painful. Should we comment on this and why everything is ok?

Reply:
Was looking at the wrong part ... Controller one has no version.

Overall maybe we should add comments in the sconstruct definitions that these MUST match binary wise. I even wonder if there is a way to test and enforce this.